sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/StateSpecs.java

```diff
@@ -186,6 +243,16 @@ private ValueStateSpec(Coder<T> coder) {
     }
 
     @Override
+    public void offerCoders(Coder[] coders) {
+      if (this.coder == null) {
+        if (coders[0] != null) {
+          //noinspection unchecked
```

Comment:
You're right, this works in IntelliJ IDEa, I'll change to an annotation on the method.